src/odemis/acq/align/autofocus.py

```diff
@@ -1003,19 +1092,25 @@ def AutoFocus(detector, emt, focus, dfbkg=None, good_focus=None, rng_focus=None,
     return f
 
 
-def AutoFocusSpectrometer(spectrograph, focuser, detectors, selector=None, streams=None):
+def AutoFocusSpectrometer(
+        spectrograph: model.Actuator,
+        focuser: model.Actuator,
+        detectors: Union[model.Detector, List[model.Detector]],
+        selector: Optional[model.Actuator] = None,
+        streams: Optional[List[Any]] = None
+) -> Dict[model.ProgressiveFuture, dict]:
```

Comment:
@pieleric I'm not sure if this is correct, this was the best I could come up with based on the docstring

Reply:
`streams` is a list of acq.stream.Stream`. Not sure how easy is it to add a dependency to the streams though. Maybe it's simpler to just write `"Stream"`.

For the return value, it's just a `model.ProgressiveFuture` . The `->` in the docstring meant that this Future will return a value.